src/main/java/com/st/utopia/agent/controller/BookingController.java

```diff
@@ -0,0 +1,203 @@
+package com.st.utopia.agent.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import com.st.utopia.agent.model.PaymentAmount;
+import com.st.utopia.agent.model.Ticket;
+import com.st.utopia.agent.model.User;
+
+/**
+ * Controller for agents using this user-facing interface
+ *
+ * @author Jonathan Lovelace
+ * @author Salem O
+ */
+@RestController
+@RequestMapping("/booking")
+public class BookingController {
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Value("${utopia.booking-service-hostName}")
+	private String hostName;
+
+	@Value("${utopia.booking-service-port}")
+	private String port;
+	
+	@Value("${utopia.booking-service-controller-root}")
+	private String root;
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for "get-all"
+	 * methods.
+	 * @param url the URL to send the REST request to
+	 * @param <T> the type we expect
+	 * @return the response the server sent
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
+				new ParameterizedTypeReference<T>() {});
+	}
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for a method call.
+	 * @param url the URL to send the REST request to
+	 * @param type of method to send the REST request to
+	 * @param HttpEntity to send in the REST request
+	 * @param <T> the type we expect
+	 * @param <U> the type of the body sent to the server
+	 * @return the response the server sent
+	 */
+	private <T, U> ResponseEntity<T> methodCall(final String url, final HttpMethod method,
+			final HttpEntity<U> body) {
+		return restTemplate.exchange(url, method, body,
+				new ParameterizedTypeReference<T>() {});
+	}
+
+	/**
+	 * Reserve a ticket for the given seat.
+	 * @param flight the flight number of the flight
+	 * @param row the row number of the seat
+	 * @param seat the seat within the row
+	 * @param user the user details
+	 * 
+	 * @return the ticket that was created by this post
+	 */
+	@PostMapping("/book/flights/{flight}/rows/{row}/seats/{seat}")
```

Comment:
The endpoints in the request-mapping annotations are not the API in the specification (because the API presented by the backend is not that API).